patterns-use-cases/payment-state-machine/payment-state-machine-java/README.md

```diff
@@ -0,0 +1,47 @@
+# Simple Payment State Machine
+
+This example shows how to build a reliable payment state machine.
+
+The state machine ensures that payments are processed once, not duplicated,
+can be revoked, and that concurrent payment requests and cancellations sort
+out consistently.
+
+The example illustrates the following aspects:
+
+- Payment requests use a token to identify payments (stripe-style)
+- Restate tracks the status of each payment request by token in internal state.
+- A payment can be cancelled, which prevents it from succeeding later, or rolls it back, if
+  it was already processed.
+- Virtual Object concurrency ensures that requests and cancellations don't produce
+  tricky race conditions.
+- Expiry of tokens is handled through Restate's internal timers.
+
+Despite the relatively few lines of code (no careful synchronization, retries, or other recovery logic),
+this application maintains a high level of consistency in the presence of concurrent external requests
+and failures.
+
+
+## Running this example
+
+Start a Restate Server: `npx restate-server`
```

Comment:
Java users perhaps don't use npx...

Reply:
But we don't have a concise command to run it otherwise: Docker or downloading the script.

Reply:
But then suggest docker or homebrew here, and then send people back to the quickstart for the other methods. Suggesting npx here seems completely wrong.